examples/combined_ODNP.py

```diff
@@ -0,0 +1,352 @@
+            'nScans', 'nEchoes', 'p90_us', 'deadtime_us', 'repetition_us', 'SW_kHz',
+            'nPoints', 'tau_adjust_us', 'deblank_us', 'tau_us', 'MWfreq', 'acq_time_ms', 'meter_power']}
+        vd_data.set_prop('acq_params',acq_params)
+        vd_data.name(T1_node_names[j])
+        myfilename = date+'_'+output_name+'.h5'
+        ph1ir_cyc = r_[0,2]
+        ph2ir_cyc = r_[0,2]
+        vd_data.setaxis('ph1',ph1ir_cyc/4)
+        vd_data.setaxis('ph2',ph2ir_cyc/4)
+        vd_data.hdf5_write(myfilename)
+        time_list.append(time.time())
+#}}}
+#{{{run enhancement
+DNP_ini_time = time.time()
+DNP_data = run_scans(nScans,0)
+DNP_thermal_done = time.time()
+time_list.append(time.time())
+time_axis_coords = DNP_data.getaxis('indirect')
+time_axis_coords[0]['start_times'] = DNP_ini_time
+# w/ struct array, this becomes time_axis_coords[0]['start_time']
+DNP_data.set_prop('start_time', DNP_ini_time)
+DNP_data.set_prop('thermal_done_time', DNP_thermal_done)
+time_axis_coords[0]['stop_times'] = DNP_thermal_done
+# w/ struct array, we can add time_axis_coords[0]['stop_time'], and the above becomes obsolete
+power_settings = zeros_like(dB_settings)
+time.strftime('%Y-%m-%d %H:%M:%S',time.localtime(time.time()))
+with power_control() as p:
+    for j, this_dB in enumerate(dB_settings):
+        print("SETTING THIS POWER",this_dB,"(",dB_settings[j-1],powers[j],"W)")
+        if j == 0:
+            retval = p.dip_lock(9.81,9.83)
+            print(retval)
+        p.set_power(this_dB)
+        for k in range(10):
+            time.sleep(0.5)
+            if p.get_power_setting() >= this_dB: break
+        if p.get_power_setting() < this_dB: raise ValueError("After 10 tries, the power has still not settled")    
+        time.sleep(5)
+        power_settings[j] = p.get_power_setting()
+        time_axis_coords[j+1]['start_times'] = time.time()
+        run_scans(nScans,j+1,DNP_data)
+        time_axis_coords[j+1]['stop_times'] = time.time()
+    this_log=p.stop_log()
+DNP_data.set_prop('stop_time', time.time())
+acq_params = {j:eval(j) for j in dir() if j in ['adcOffset', 'carrierFreq_MHz', 'amplitude',
+    'nScans', 'nEchoes', 'p90_us', 'deadtime_us', 'repetition_us', 'SW_kHz',
+    'nPoints', 'tau_adjust_us', 'deblank_us', 'tau_us', 'nPhaseSteps', 'MWfreq', 'power_settings','pul_prog']}
+DNP_data.set_prop('acq_params',acq_params)
+myfilename = date+'_'+output_name+'.h5'
+DNP_data.chunk('t',['ph1','t2'],[4,-1])
+DNP_data.setaxis('ph1',r_[0,1,2,3]/4)
+time_list.append(time.time())
+time_array = array(time_list)
+while save_file:
+    try:
+        print("SAVING FILE...")
+        DNP_data.hdf5_write(myfilename)        
+        print("FILE SAVED")
+        print("Name of saved enhancement data", DNP_data.name())
+        print("shape of saved enhancement data", ndshape(DNP_data))
+        save_file=False
+    except Exception as e:
+        print(e)
+        print("EXCEPTION ERROR - FILE MAY ALREADY EXIST.")
+        save_file = False
```

Comment:
here, set save_file to False.  Add `time.sleep(3)`.  Then, it would keep trying until the "try" part succeeded or you pressed ctrl-C or killed the process.

The point would be that you could rename the old file in a separate window -- then it could save with the file name you had given, and you would not lose your data.

You would want to edit the EXCEPTION ERROR message here to explain what's going on, though, because you are deliberately setting up an infinite loop.